TaskfMRIAnalysis/TaskfMRIAnalysis.sh

```diff
@@ -39,56 +39,92 @@
+    echo "
+$log_ToolName: Run TaskfMRIAnalysis pipeline for a subject. Pipeline will run Level1 (scan-level) analyses, and Level2 (single subject-level) analysis as specified.
+
+Usage: $log_ToolName arguments...
+[ ] = optional; < > = user supplied value
+"
+    #automatic argument descriptions
+    opts_ShowArguments
+    
+    #do not use exit, the parsing code takes care of it
 }
 
-# Allow script to return a Usage statement, before any other output or checking
-if [ "$#" = "0" ]; then
-    show_usage
-    exit 1
-fi
-
 # ------------------------------------------------------------------------------
-#  Check that HCPPIPEDIR is defined and Load Function Libraries
+#  Parse Command Line Options
 # ------------------------------------------------------------------------------
 
-if [ -z "${HCPPIPEDIR}" ]; then
-  echo "${script_name}: ABORTING: HCPPIPEDIR environment variable must be set"
-  exit 1
-fi
+#arguments to opts_Add*: switch, variable to set, name for inside of <> in help text, description, [default value if AddOptional], [compatibility flag, ...]
+#help info for option gets printed like "--foo=<$3> - $4"
+opts_AddMandatory '--study-folder' 'Path' '/path/to/study/folder' "directory containing imaging data for all subjects"
+opts_AddMandatory '--subject' 'Subject' 'SubjectID' ""
+opts_AddMandatory '--lvl1tasks' 'LevelOnefMRINames' 'tfMRI_TaskName_RL@tfMRI_TaskName_LR' "List of subdirectories containing each time series image for the TaskName task. Multiple scan subdirectories should be provided as a single string separated by '@' character." #Assumes these subdirectories are located in the SubjectID/MNINonLinear/Results directory. Also assumes that timeseries image filename begins with this string.
+opts_AddOptional '--lvl1fsfs' 'LevelOnefsfNames' 'DesignName_TaskName_RL@DesignName_TaskName_LR' "List of prefixes in the design.fsf filenames for each time series image for the TaskName task. Should contain same number of design files as time series images in --lvl1tasks option. (N-th design will be used for N-th time series image.) Separate multiple design files by '@' character. If no value is passed to --lvl1fsfs, the value will be set to the same list passed to --lvl1tasks."
+opts_AddOptional '--lvl2task' 'LevelTwofMRIName' 'tfMRI_TaskName' "Name of Level2 subdirectory in which all Level2 feat directories are written for TaskName. Default is 'NONE', which means that no Level2 analysis will run." 'NONE'
+opts_AddOptional '--lvl2fsf' 'LevelTwofsfName' 'DesignName_TaskName' "Prefix of design.fsf filename for the Level2 analysis for TaskName. If no value is passed to --lvl2fsf, the value will be set to the same list passed to --lvl2task."
+opts_AddOptional '--lowresmesh' 'LowResMesh' 'integer' "Default=32, which is appropriate for HCP minimal preprocessing pipeline outputs" '32'
+opts_AddOptional '--grayordinatesres' 'GrayordinatesResolution' 'integer' "Default=2mm, which is appropriate for HCP minimal preprocessing pipeline outputs" '2'
```

Comment:
Maybe for this, and the other variables like it, we should do: `'integer (mm)' "Default=2, ...`"

Reply:
I would probably put the "mm" into the explanation and leave it saying `[--grayordinates=<integer>]`, but yes, the default should probably say `2`, without "mm" on the end.

Reply:
Actually, isn't the main other choice for grayordinates 1.6?  It probably shouldn't say integer then...

Reply:
greyordinates are not necessarily integer resolution.

Reply:
Hmmm... since the variable is ultimately used to specify the name of file in `wb_command -cifti-create-dense-timeseries`, I guess we should technically say that it is a string. If we call it a float, people might enter 2.0 or 1.60 and expect it to work.

Reply:
A possibility is to say `--grayordinatesres=<2 or 1.6>`, or `--grayordinatesres=<number> - 2 (default) or 1.6 ...`, assuming we don't plan to make another grayordinates resolution soon.  Not future-proof, but fairly clear.

For reference, here is what I did in PostFreeSurfer (mandatory, not optional, but we expect the user to use Examples/Scripts/...Batch.sh to launch it), it may not be perfect, but we also may not need to be so picky:
```
   --grayordinatesres=<number> - usually '2', resolution of grayordinates to
        use
```

Reply:
Other species have other resolutions.

Reply:
Changed to:
opts_AddOptional '--grayordinatesres' 'GrayordinatesResolution' 'integer_as_string' "Use value in 'Atlas_ROIs' filename; Default=2mm ...

Reply:
This should not use the word "integer", 1.6 is a likely value.  "2mm" is not what the default is, the default is "2", and we should not make users think that the argument should contain the characters "mm".

Reply:
Oops! Yeah, I change it to:
'number_as_string' "Value (in mm) from 'Atlas_ROIs' filename; Default='2'

Reply:
I don't know that "number_as_string" is any clearer to users than "number" (bets on whether someone tries "--grayordinatesres=two"?), and I think it looks awkward.

Reply:
That's a valid point. I'll put 'number'. If anyone enters '2.0' and it doesn't run, we'll just need to leave it to them to debug why.